docs/2_3_fmu_distribution.adoc

```diff
@@ -196,16 +196,31 @@ The library names are defined in the `modelIdentifier` attribute of elements `fm
-All C source file names that need to be defined in a compiler directive have to be defined in the XML file under structure `<ModelExchange><SourceFiles>` and `<CoSimulation><SourceFiles>`.
+Exactly those source file names that need to be defined in a compiler directive have to be defined in the XML file under structure `<ModelExchange><SourceFiles>` and `<CoSimulation><SourceFiles>`.
 These files may include other files.
 `#include` directive with `""` should be used for header-filers distributed in the FMU instead of using `<...>`.
-If default options of the compiler are sufficient, it might then be possible to automatically process such source code FMUs.
-An exporting tool should give documentation on how to build an executable, either via a documentation file and/or via a template makefile for a particular platform, from which a user can construct the makefile for his/her target platform.
+
+If necessary, further information on include search paths, preprocessor definitions and additional static libraries have to be defined in the XML file under `<ModelExchange>` and `<CoSimulation>` (elements `<IncludeDirectories>`, `<PreProcessorDefinitions>` and `<Libraries>`). 
+If default options of the compiler are sufficient, it should then be possible to automatically process such source code FMUs.
+In some cases, the information provided by the FMU might not be sufficient. In these cases, an exporting tool should give documentation on how to build an executable, either via a documentation file and/or via a template makefile for a particular platform, from which a user can construct the makefile for his/her target platform.
 This documentation should be stored in the `documentation` directory, possibly with a link to the template makefile (stored in the `sources` directory).
 _[As template makefile, CMake (http://www.cmake.org), a cross- platform, open-source build system might be used.]_ +
+
+For Source Code FMUs, the importing tool must compile the source files that are listed in the `ModelDescription.xml` under the structure `<ModelExchange/CoSimulation><SourceFiles>`. Exactly those files must be considered that are valid for the corresponding platform. This can be read from the `platform` attribute of each `<ModelExchange/CoSimulation><SourceFiles><File>` element. If no platform information is specified for a source file, the file is assumed to be valid for all supported platforms. The supported platforms can be read from the `platformSupportLimitedTo` attribute of the `SourceFiles` element. If the `platformSupportLimitedTo` attribute is not provided, all platforms are supported (or at least the exporting tool knows of no restrictions).
+
+If an FMU contains source files in its sources folder, but the `SourceFiles` element is not available in the XML file, all source files should be compiled.
```

Comment:
This sentence should be removed. As we said, a source code FMU should always list the SourceFiles element.

Reply:
It would make life easier for FMU exporting Modelica tools if - for the case that all source files need to be compiled - the then redundant listing of the SourceFiles element - can be skipped. See also https://github.com/modelica/ModelicaSpecification/issues/2145 for the corresponding Modelica specification issue to improve generation of source code FMUs.

Reply:
@beutlich the Modelica tool anyway has to traverse the list of files to put them in the zip, to also list them in the xml file cannot be that much extra work. It would simplify for importers in that there is only one way to specify the list of sources to compile and there is no need to specify what should happen for different file extensions (now when we allow more than one language).

Reply:
> the Modelica tool anyway has to traverse the list of files to put them in the zip

Do not think so. A Modelica tool simply can pack an entire directory structure as sources w/o traversing it.

But I agree, that making the source file dependencies explicit helps build environments. So, fine for me.